src/main/java/appeng/fluids/parts/PartSharedFluidBus.java

```diff
@@ -0,0 +1,209 @@
+	private boolean lastRedstone;
+
+	public PartSharedFluidBus( ItemStack is )
+	{
+		super( is );
+	}
+
+	@Override
+	public void upgradesChanged()
+	{
+		this.updateState();
+	}
+
+	@Override
+	public void onNeighborChanged( IBlockAccess w, BlockPos pos, BlockPos neighbor )
+	{
+		this.updateState();
+		if( this.lastRedstone != this.getHost().hasRedstone( this.getSide() ) )
+		{
+			this.lastRedstone = !this.lastRedstone;
+			if( this.lastRedstone && this.getRSMode() == RedstoneMode.SIGNAL_PULSE )
+			{
+				this.doBusWork();
+			}
+		}
+	}
+
+	private void updateState()
+	{
+		try
+		{
+			if( !this.isSleeping() )
+			{
+				this.getProxy().getTick().wakeDevice( this.getProxy().getNode() );
+			}
+			else
+			{
+				this.getProxy().getTick().sleepDevice( this.getProxy().getNode() );
+			}
+		}
+		catch( final GridAccessException e )
+		{
+			// :P
+		}
+	}
+
+	@Override
+	public boolean onPartActivate( final EntityPlayer player, final EnumHand hand, final Vec3d pos )
+	{
+		if( !player.isSneaking() )
+		{
+			if( Platform.isClient() )
+			{
+				return true;
+			}
+
+			Platform.openGUI( player, this.getHost().getTile(), this.getSide(), GuiBridge.GUI_BUS_FLUID );
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public void getBoxes( IPartCollisionHelper bch )
+	{
+		bch.addBox( 6, 6, 11, 10, 10, 13 );
+		bch.addBox( 5, 5, 13, 11, 11, 14 );
+		bch.addBox( 4, 4, 14, 12, 12, 16 );
+	}
+
+	protected TileEntity getConnectedTE()
+	{
+		TileEntity self = this.getHost().getTile();
+		return this.getTileEntity( self, self.getPos().offset( this.getSide().getFacing() ) );
+	}
+
+	private TileEntity getTileEntity( final TileEntity self, final BlockPos pos )
+	{
+		final World w = self.getWorld();
+
+		if( w.getChunkProvider().getLoadedChunk( pos.getX() >> 4, pos.getZ() >> 4 ) != null )
+		{
+			return w.getTileEntity( pos );
+		}
+
+		return null;
+	}
+
+	@Override
+	public IItemHandler getInventoryByName( final String name )
+	{
+		if( name.equals( "config" ) )
+		{
+			return this.getConfig();
+		}
+
+		return super.getInventoryByName( name );
+	}
+
+	protected int calculateAmountToSend()
+	{
+		double amount = this.getChannel().transferFactor();
+		switch( this.getInstalledUpgrades( Upgrades.SPEED ) )
+		{
+			case 4:
+				amount = amount * 1.5;
+			case 3:
+				amount = amount * 2;
+			case 2:
+				amount = amount * 4;
+			case 1:
+				amount = amount * 8;
+			case 0:
+			default:
+				return MathHelper.floor( amount );
+		}
+	}
+
+	@Override
+	public void readFromNBT( NBTTagCompound extra )
+	{
+		super.readFromNBT( extra );
+		this.getConfig().readFromNBT( extra, "config" );
+	}
+
+	@Override
+	public void writeToNBT( NBTTagCompound extra )
+	{
+		super.writeToNBT( extra );
+		this.getConfig().writeToNBT( extra, "config" );
+	}
+
+	public AppEngInternalAEInventory getConfig()
+	{
+		return this.config;
+	}
+
+	protected IFluidStorageChannel getChannel(){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)